src/components/chatX/MessagesDisplay.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+
+import { watchIncomingMessage, unwatchIncomingMessage } from '../../listeners/messageListeners';
+
+import '../../styles/chatx.css';
+
+class MessagesDisplay extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  // This component can be mounted when a room is already selected
+  // If so, start waching right away
+  componentWillMount() {
+    if (this.props.currentRoomId) {
+      this.props.actions.watchIncomingMessage(this.props.currentRoomId);
```

Comment:
`watchIncomingMessage` and `unwatchIncomingMessage` could have been avoided here if messages were listened globally (by redux) upon room creation.
`<MessagesDisplay />` would have taken a `roomId` parameter, and the `mapStateToProps` would have looked like:

```javascript
function mapStateToProps(state, props) {
  return {
    messages: state.rooms[props.roomId].messages,
    currentRoomId: state.rooms.current
  };
```